kratos/processes/find_global_nodal_elemental_neighbours_process.h

```diff
@@ -0,0 +1,449 @@
+                }
+            }
+
+            //here communicate non local data
+            //compute communication plan
+            std::vector<int> send_list;
+            send_list.reserve( non_local_map.size() );
+            for(auto& it : non_local_map)
+                send_list.push_back( it.first );
+
+            std::sort(send_list.begin(), send_list.end());
+            auto colors = MPIColoringUtilities::ComputeCommunicationScheduling(send_list, mrComm);
+
+            //finalize computation of neighbour ids on owner nodes
+            non_local_map_type recv_map;
+            for(int color : colors)
+            {
+                if(color >= 0)
+                {
+                    //recev the global neighbours as computed on color
+                    recv_map[color] = SendRecv(non_local_map[color], color, color );
+
+                    for(auto& item : recv_map[color])
+                    {
+                        auto& recv_node = mr_model_part.Nodes()[item.first]; 
+                        auto& neighbours = recv_node.GetValue(NEIGHBOUR_ELEMENTS);
+                        for(auto& gp : item.second.GetContainer())
+                            neighbours.push_back(gp);
+                    }
+                }
+            } //after this loop is finished neighbours are ok for the owner nodes
+
+            //fill back the recv_map with the updated information
+            for(int color : colors)
+            {
+                if(color >= 0)
+                {
+                    for(auto& item : recv_map[color])
+                    {
+                        //item.first contains the id of the node
+                        //item.second contains the list of neighbours
+                        auto& recv_node = mr_model_part.Nodes()[item.first]; 
+                        item.second = recv_node.GetValue(NEIGHBOUR_ELEMENTS);
+                    }
+
+                    //obtain the final list of neighbours for nodes owned on color
+                    auto final_gp_map = SendRecv(recv_map[color], color, color );
+
+                    //update the local database
+                    for(auto& item : final_gp_map)
+                    {
+                        auto& recv_node = mr_model_part.Nodes()[item.first]; 
+                        recv_node.GetValue(NEIGHBOUR_ELEMENTS) = item.second;
+                    }
+                }
+            }
+        }
+
+        auto constructor_functor =  Kratos::ComputeNeighbourListFunctor<
+                            ModelPart::NodesContainerType, 
+                            Variable< GlobalPointersVector< Element > >
+                            > (rNodes, NEIGHBOUR_ELEMENTS);
+
+        GlobalPointerCommunicator<Element> pointer_comm(mrComm, constructor_functor );
+        auto id_proxy = pointer_comm.Apply(
+                [](GlobalPointer<Element>& gp){return gp->Id();}
+        );
+
+        #pragma omp parallel for
```

Comment:
Not a complain, but if we are openmp this one we could parallelize some others. (117, 235, not sure but 260 as well)

Reply:
done